docs/source/filters.rst

```diff
@@ -517,15 +515,40 @@ string).
            pattern: '</([^>]*)>'
            repl: '<END OF TAG \1>'
 
-If you want to enable certain flags (e.g. ``re.MULTILINE``) in the
-call, this is possible by inserting an "inline flag" documented in
-`flags in re.compile`_, here are some examples:
+``re.findall`` can be used to find all non-overlapping matches of a
+regular expression. Each match is output on its own line. The following
+example applies the filter twice:
+
+1. It uses a group (``()``) and back-reference (``\1``) to extract a
+   date from the input string.
+2. It breaks the numbers in the date out into separate lines.
+
+If ``repl`` is not specified, the full match will be included in the output.
+
+.. code:: yaml
+
+   url: https://example.com/regex-findall.html
+   filter:
+       - re.findall:
+           pattern: 'The next draw is on (\d\d\d\d-\d\d-\d\d).'
+           repl: '\1'
+       - re.findall: '[0-9]+'
+
+Note: When using HTML or XML, it is usually better to use CSS selectors or
+XPATH expressions. HTML and XML cannot be parsed properly using regular
+expressions. If the CSS selector or XPATH cannot provide the targeted
```

Comment:
Could add a link to https://stackoverflow.com/a/1732454/1047040 on "cannot be parsed" for extra entertainment.

Reply:
It is a classic :)